spr/src/message.rs

```diff
@@ -366,4 +407,209 @@ Reviewers:    a, b, c
             .into()
         );
     }
+
+    // -----------------------------------------------------------------
+    // build_message*() tests
+
+    #[test]
+    fn test_build_message_just_title() {
+        assert_eq!(
+            build_message(
+                &parse_message(
+                    r#"test: just title
+
+"#,
+                    MessageSection::Title,
+                ),
+                &[
+                    MessageSection::Title,
+                    MessageSection::Summary,
+                ],
+            ),
+
+            "test: just title\n"
+        );
+    }
+
+    // -------------------------------------------------
+    #[test]
+    fn test_build_message_just_title_and_summary() {
+        assert_eq!(
+            build_message(
+                &parse_message(
+                    r#"Just title and summary
+
+Notice: not a trailer
+
+More summary here
+
+"#,
+                    MessageSection::Title,
+                ),
+                &[
+                    MessageSection::Title,
+                    MessageSection::Summary,
+                ],
+            ),
+            r#"Just title and summary
+
+Notice: not a trailer
+
+More summary here
+"#,
+        );
+    }
+
+    // -------------------------------------------------
+    #[test]
+    fn test_build_message_no_blank_between_title_and_summary() {
+        assert_eq!(
+            build_message(
+                &parse_message(
+                    r#"No blank line between title and summary
+Summary"#,
+                    MessageSection::Title,
+                ),
+                &[
+                    MessageSection::Title,
+                    MessageSection::Summary,
+                ],
+            ),
+            r#"No blank line between title and summary
+
+Summary
+"#,
+        );
+    }
+
+    // -------------------------------------------------
+    #[test]
+    fn test_build_message_just_title_and_known_trailer() {
+        assert_eq!(
+            build_message(
+                &parse_message(
+                    r#"Just title and known trailer
+
+ Test-Plan: foobar
+"#,
+                    MessageSection::Title,
+                ),
+                &[
+                    MessageSection::Title,
+                    MessageSection::Summary,
+                    MessageSection::TestPlan,
+                ],
+            ),
+            r#"Just title and known trailer
+
+Test-Plan: foobar
+"#,
+        );
+    }
+
+    // -------------------------------------------------
+    #[test]
+    fn test_build_message_title_summary_known_trailers() {
+        assert_eq!(
+            build_commit_message(
+                &parse_message(
+                    r#"test: title, summary and regular sections
+
+Summary: not a trailer
+
+ http://example.com/foo
```

Comment:
Q: Is the single space before `http` deliberate?

Reply:
yes; will add both with and without heading spaces, just in case. In the original implementation, such line would be deleted with and without heading spaces 🙄